### ISSUE TITLE
No mock implementation for remove_delayed_selection

### ISSUE THREAD

**Issue by olliebennett** [justinweiss/resque_unit#41]
The `remove_delayed_selection` method was added to `resque-scheduler` in [January 2014](https://github.com/resque/resque-scheduler/blob/master/HISTORY.md#240-2014-01-14) (and subsequently improved in [December 2014](https://github.com/resque/resque-scheduler/blob/master/HISTORY.md#310-2014-12-21)).

Neither of the following uses are currently mocked by this gem:

```ruby
Resque.remove_delayed_selection { |a| a.first == 'bar' }
```

```ruby
Resque.remove_delayed_selection(SomeJob) do |a|
    a.first == 'foo'
end
```

For more info / examples, find the `remove_delayed_selection` tests in `resque-scheduler`'s [delayed_queue_test.rb](https://github.com/resque/resque-scheduler/blob/master/test/delayed_queue_test.rb).

Perhaps I could attempt to implement this (if you don't have time), especially if you had any pointers for where to start. Was thinking it would be similar to work done by @archit in https://github.com/justinweiss/resque_unit/commit/3dd4ae2159e5c916e6af3bfd5068a945e0dc12ea?

**Comment by justinweiss** [comment]
> Perhaps I could attempt to implement this (if you don't have time), especially if you had any pointers for where to start. Was thinking it would be similar to work done by @archit in 3dd4ae2?

Thanks, I'd definitely appreciate the contribution! Following @archit's pattern should be fine.

**Comment by justinweiss** [comment]
Hey, I just released a beta that should fix the issue -- if you want, try `v1.0.0.beta.1`.
